R/args.R

```diff
@@ -147,15 +149,36 @@ SampleArgs <- R6::R6Class(
       self$thin <- thin
       self$max_depth <- max_depth
       self$metric <- metric
-      # self$metric_file <- character()
+      self$inv_metric <- inv_metric
+      if(!is.null(inv_metric)) {
+        if(!is.null(metric_file)) {
+          stop("Only one of inv_metric and metric_file can be non-NULL")
+        }
+
+        # wrap inv_metric in list if not in one
+        if(!is.list(inv_metric)) {
+          inv_metric = list(inv_metric)
+        }
+
+        # write all inv_metrics to disk
+        inv_metric_paths <-
+          tempfile(
+            pattern = paste0("inv_metric-", 1:length(inv_metric), "-"),
+            tmpdir = cmdstan_tempdir(),
+            fileext = ".json"
+          )
+        for(i in seq_along(inv_metric_paths)) {
+          write_stan_json(list(inv_metric = inv_metric[[i]]), inv_metric_paths[i])
+        }
+
+        self$metric_file <- inv_metric_paths
+      } else {
+        self$metric_file <- sapply(metric_file, absolute_path)
+      }
```

Comment:
It seems that tests are still failing (not your metric tests) because now _not_ including `inv_metric` or `metric_file` results in an error (that's the hazard of just running the metric tests!). So allowing for `NULL` in `validate_metric_file` wasn't actually enough. I think what will do the trick is changing the `else` clause above to this:

```r
      } else if (!is.null(metric_file)) {
        self$metric_file <- sapply(metric_file, absolute_path)
      }
```